azurerm/internal/services/loganalytics/log_analytics_storage_insights_resource.go

```diff
@@ -0,0 +1,211 @@
+	}
+
+	workspaceId := d.Get("workspace_id").(string)
+	id := parse.NewLogAnalyticsStorageInsightsId(resourceGroup, workspaceId, name)
+
+	if d.IsNewResource() {
+		existing, err := client.Get(ctx, resourceGroup, id.WorkspaceName, name)
+		if err != nil {
+			if !utils.ResponseWasNotFound(existing.Response) {
+				return fmt.Errorf("checking for present of existing Log Analytics Storage Insights %q (Resource Group %q / workspaceName %q): %+v", name, resourceGroup, id.WorkspaceName, err)
+			}
+		}
+		if existing.ID != nil && *existing.ID != "" {
+			return tf.ImportAsExistsError("azurerm_log_analytics_storage_insights", *existing.ID)
+		}
+	}
+
+	parameters := operationalinsights.StorageInsight{
+		StorageInsightProperties: &operationalinsights.StorageInsightProperties{
+			StorageAccount: expandArmStorageInsightConfigStorageAccount(storageAccountId, storageAccountKey),
+		},
+		Tags: tags.Expand(d.Get("tags").(map[string]interface{})),
+	}
+
+	if _, ok := d.GetOk("table_names"); ok {
+		parameters.StorageInsightProperties.Tables = utils.ExpandStringSlice(d.Get("table_names").(*schema.Set).List())
+	}
+
+	if _, ok := d.GetOk("blob_container_names"); ok {
+		parameters.StorageInsightProperties.Containers = utils.ExpandStringSlice(d.Get("blob_container_names").(*schema.Set).List())
+	}
+
+	if _, err := client.CreateOrUpdate(ctx, resourceGroup, id.WorkspaceName, name, parameters); err != nil {
+		return fmt.Errorf("creating/updating Log Analytics Storage Insights %q (Resource Group %q / workspaceName %q): %+v", name, resourceGroup, id.WorkspaceName, err)
+	}
+
+	d.SetId(id.ID(subscriptionId))
+	return resourceArmLogAnalyticsStorageInsightsRead(d, meta)
+}
+
+func resourceArmLogAnalyticsStorageInsightsRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).LogAnalytics.StorageInsightsClient
+	ctx, cancel := timeouts.ForRead(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id, err := parse.LogAnalyticsStorageInsightsID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	// Need to pull this from the config since the API does not return this value
+	storageAccountKey := d.Get("storage_account_key").(string)
+
+	resp, err := client.Get(ctx, id.ResourceGroup, id.WorkspaceName, id.Name)
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			log.Printf("[INFO] Log Analytics Storage Insights %q does not exist - removing from state", d.Id())
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("retrieving Log Analytics Storage Insights %q (Resource Group %q / workspaceName %q): %+v", id.Name, id.ResourceGroup, id.WorkspaceName, err)
+	}
+
+	d.Set("name", id.Name)
+	d.Set("resource_group_name", id.ResourceGroup)
+	d.Set("workspace_id", id.WorkspaceID)
+
+	if props := resp.StorageInsightProperties; props != nil {
+		d.Set("blob_container_names", utils.FlattenStringSlice(props.Containers))
+		d.Set("storage_account_id", props.StorageAccount.ID)
```

Comment:
theres a crash here if `props.StorageAccount` is nil:

```suggestion
		storageAccountId := ""
		if props.StorageAccount != nil && props.StorageAccount.ID != nil {
			storageAccountId = *props.StorageAccount.ID
		}
		d.Set("storage_account_id", storageAccountId)
```